user.tf

```diff
@@ -42,3 +42,19 @@ resource "postgresql_default_privileges" "readonly_privileges" {
   object_type = "table"
   privileges  = ["SELECT"]
 }
+
+resource "postgresql_grant" "readonly_grants" {
+  for_each = merge(
+    [
+      for key, user in local.postgres_users : {
+        for db in var.databases : "${db}/${key}" => user if user.readonly == true
+      }
+    ]...
+  )
+
+  database    = google_sql_database.pararius_office.name
+  role        = postgresql_role.readonly_roles[each.value].name
+  schema      = "public"
+  object_type = "table"
+  privileges  = []
```

Comment:
https://registry.terraform.io/providers/cyrilgdn/postgresql/latest/docs/resources/postgresql_grant#privileges

I doubt an empty array is desired :wink:

I'm guessing our "fake" super user doesn't know how to fetch these privileges, therefor defaulting to `[]`. It remains imperative to set the proper desired privileges.

Reply:
But is it? As far as I can tell, privileges here are ones granted in addition to the default privileges defined above.
https://www.postgresql.org/docs/13/sql-alterdefaultprivileges.html

Perhaps it's better to just specify grants and remove the default privileges...